lib/cloud_controller/legacy_api/legacy_bulk.rb

```diff
@@ -70,7 +70,8 @@ def bulk_apps
       bulk_token = MultiJson.load(params.fetch('bulk_token'))
       last_id = Integer(bulk_token['id'] || 0)
 
-      if last_id > 0
+      max_process_id = App.db["Select max(id) from processes p WHERE p.state ='STARTED' AND p.diego IS FALSE"].first
```

Comment:
What happens when a new app is added, the max changes.
Why not just send back a sentinel instead like a negative number or the actual MAX value of id.
Sequel can do this for you too via App.max(:id).where(state: 'STARTED').where(diego: FALSE)